soundcard/pulseaudio.py

```diff
@@ -351,11 +354,13 @@ def record(self, numframes):
 
         """
 
-        captured_frames = 0
-        captured_data = []
+        if self.last_record + time.time() > numframes / self._samplerate:
+            self.captured_data = []
+            self.captured_frames = 0
+
```

Comment:
Have you tested this? I believe that pulseaudio will buffer audio data for a while even if you don't read it immediately. if I read the documentation correctly, it will raise some kind of error before discarding audio data (probably setting `data_ptr` to `NULL` when calling [`_pa_stream_peek`](https://freedesktop.org/software/pulseaudio/doxygen/stream_8h.html#ac2838c449cde56e169224d7fe3d00824)).